File: src/main/java/skywolf46/extrautility/areas/IArea.java
package skywolf46.extrautility.areas;

import org.bukkit.Location;
import org.bukkit.World;

public interface IArea {
    int getPoint();

    boolean isInArea(Location loc);

    IArea create(Location[] locs);

    World getWorld();

    Location[] getPoints();
}

File: src/main/kotlin/skywolf46/extrautility/abstraction/IArea.kt
package skywolf46.extrautility.abstraction

import org.bukkit.Location
import org.bukkit.World

interface IArea {
    fun isInArea(x: Location): Boolean

    fun point(): Int

    fun create(x: Array<Location>): IArea?

    fun world(): World

    fun locations(): Array<Location>
}